src/Simplex/Messaging/TMap.hs

```diff
@@ -56,6 +57,10 @@ delete :: Ord k => k -> TMap k a -> STM ()
 delete k m = modifyTVar' m $ M.delete k
 {-# INLINE delete #-}
 
+filterWithKey :: (k -> a -> Bool) -> TMap k a -> STM ()
```

Comment:
this is not correct naming - "filter" is supposed to return a new map, not modify - better to do it explicitely, without this function.